Compiler warning on `create_library`

```
In function ‘strncpy’,
    inlined from ‘set_depot_path’ at /home/kc/.julia/packages/PackageCompiler/0A6ug/src/julia_init.c:61:5:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10: warning: ‘__builtin_strncpy’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   96 |       __glibc_objsize (__dest));
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kc/.julia/packages/PackageCompiler/0A6ug/src/julia_init.c: In function ‘set_depot_path’:
/home/kc/.julia/packages/PackageCompiler/0A6ug/src/julia_init.c:47:24: note: length computed here
   47 |     int root_dir_len = strlen(root_dir);
      |                        ^~~~~~~~~~~~~~~~
```

cc @kmsquire

--- a/src/julia_init.c
+++ b/src/julia_init.c
@@ -55,11 +55,11 @@ void set_depot_path(char *sysimage_path)
 #endif
     int depot_dir_len = strlen(depot_subdir);
 
-    int full_path_len = root_dir_len + depot_dir_len;
-    char *dir = malloc(full_path_len + 1);
+    int full_path_len = root_dir_len + depot_dir_len + 1;
+    char *dir = malloc(full_path_len);
 
-    strncpy(dir, root_dir, root_dir_len);
-    strncpy(dir + root_dir_len, depot_subdir, depot_dir_len);
+    strncpy(dir, root_dir, full_path_len);
+    strncpy(dir + root_dir_len, depot_subdir, full_path_len - root_dir_len);
     dir[full_path_len] = '\0';
 
 #ifdef _WIN32

